pkg/scalers/scaler_test.go

```diff
@@ -85,3 +85,38 @@ func TestGetMetricTarget(t *testing.T) {
 		})
 	}
 }
+
+func TestRemoveIndexFromMetricName(t *testing.T) {
+	cases := []struct {
+		scalerIndex                          int
+		metricName                           string
+		expectedMetricNameWithoutIndexPrefix string
+		isError                              bool
+	}{
+		// Proper input
+		{scalerIndex: 0, metricName: "s0-metricName", expectedMetricNameWithoutIndexPrefix: "metricName", isError: false},
+		// Incorrect index prefix
+		{scalerIndex: 1, metricName: "s0-metricName", expectedMetricNameWithoutIndexPrefix: "", isError: true},
+		// Incorrect index prefix
+		{scalerIndex: 0, metricName: "0-metricName", expectedMetricNameWithoutIndexPrefix: "", isError: true},
+		// No index prefix
+		{scalerIndex: 0, metricName: "metricName", expectedMetricNameWithoutIndexPrefix: "", isError: true},
```

Comment:
Could you add a test case with index > 9? I mean, it's not common but it's a possibility that same ScaledObject have 10 or more scalers. I'm not sure if current implementation supports it